plugins/modules/web_infrastructure/jenkins_job.py

```diff
@@ -256,9 +266,7 @@ def has_state_changed(self, status):
         if self.enabled is None:
             return False
 
-        if ((self.enabled is False and status != "disabled") or (self.enabled is True and status == "disabled")):
-            return True
-        return False
+        return (self.enabled is False and status != "disabled") or (self.enabled is True and status == "disabled")
```

Comment:
```suggestion
        return (self.enabled is False and status != "disabled") or (self.enabled is True and status == "disabled")
```
it's usually much easier to review things that are related to each other:)
IMO it'd be better to split a main idea and refactoring into 2 PRs:) I'm speaking generally now, no need to split this PR